### DISCUSSION TITLE
Expose as Go Package

### DISCUSSION THREAD

**Discussion by ytsruh** [faroe-archive/faroe#9] [Ideas]
I was having a play around with this last night and its a really nice (& simple) implementation. I wondered if you'd be open to exposing the underlying Go code as a package that could be pulled into a Go project?

I guess the flags could be swapped out for a config struct or similar?

**Comment by pilcrowonpaper** [comment]
I did consider it, but I'm not sure how useful it'd be. You'd still have to have a separate SQLite database just for Faroe, unless we add adapters, which I'm not very keen on doing. I guess it could also be used for configuring Faroe's behavior? Honestly might be easier if you just fork Faroe, make some changes, and serve that instead.

**Reply by ytsruh** [reply]
Fair. I was thinking of using an existing DB & server (if available) and allow them through a config. However on balance I think your take is probably right. It would like add unnecessary maintenance to the project & diverge from the core idea as well.

I'll have a play around with a fork as you suggested. Thanks a lot
